plugins/inputs/cpu/cpu.go

```diff
@@ -113,6 +122,19 @@ func (c *CPUStats) Gather(acc telegraf.Accumulator) error {
 	return err
 }
 
+func (c *CPUStats) initCPUInfo() {
+	cpuInfo, err := cpuUtil.Info()
+	if err == nil {
+		c.coreID = cpuInfo[0].CoreID != ""
+		c.physicalID = cpuInfo[0].PhysicalID != ""
+
+		c.cpuInfo = make(map[string]cpuUtil.InfoStat)
+		for _, ci := range cpuInfo {
+			c.cpuInfo[fmt.Sprintf("cpu%d", ci.CPU)] = ci
+		}
+	}
+}
+
```

Comment:
Please call this function `Init() error` and telegraf will automatically execute it after the config options are filled in.
```suggestion
func (c *CPUStats) Init() error {
	cpuInfo, err := cpuUtil.Info()
	if err == nil {
		c.coreID = cpuInfo[0].CoreID != ""
		c.physicalID = cpuInfo[0].PhysicalID != ""

		c.cpuInfo = make(map[string]cpuUtil.InfoStat)
		for _, ci := range cpuInfo {
			c.cpuInfo[fmt.Sprintf("cpu%d", ci.CPU)] = ci
		}
	}
	return nil
}

```

Reply:
I have also changed the function slightly to not collect Info if CoreTags are not enabled and to log error returned from `cpuUtil.Info`

Thanks for the tip about `Init() error`